src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -267,6 +273,45 @@ class ExposureSubmissionCoordinator: NSObject, ExposureSubmissionCoordinating, R
 		)
 	}
 
+	private func createWarnOthersViewController() -> ExposureSubmissionWarnOthersViewController {
+		var vc: ExposureSubmissionWarnOthersViewController?
+
+		// ugly but works for the moment
+		// refactoring more of the coordinator-logic to facilitate combine would help
+		let group = DispatchGroup()
+		group.enter()
+
+		// get list of supported countries and create warn others view controller
+		appConfigurationProvider
+			.supportedCountries()
+			.sink(
+				receiveCompletion: { _ in
+					// no op
+				},
+				receiveValue: { countries in
+					vc = ExposureSubmissionWarnOthersViewController(
+						supportedCountries: countries,
+						onPrimaryButtonTap: { [weak self] isLoading in
+							self?.model.exposureSubmissionService.isSubmissionConsentGiven = true
+							self?.model.exposureSubmissionService.getTemporaryExposureKeys { error in
+								isLoading(false)
+								if let error = error {
+									self?.showErrorAlert(for: error)
+								} else {
+									self?.showThankYouScreen()
+								}
+							}
+						}
+					)
+					group.leave()
+				})
+			.store(in: &subscriptions)
+
+		group.wait()
+		// swiftlint:disable:next force_unwrapping
+		return vc!
```

Comment:
Instead of force unwrap and disable the linter, you could use a guard with fatalError. This way you could also provide an meaningful error message.

Reply:
If a force unwrap fails it's a developer error. No need to adopt mechanisms that are user-targeted here.

Reply:
What aredoyou mean by user-targeted?

Reply:
`fatalError` is to create proper crash logs. This code here would 'just fail' but during local and CI testing. Given a proper test coverage (✅  and still improving), there is no need to write extra guards.

Reply:
Sorry, I don't agree. If it fails on the CI, it would be a good help for the developer in charge to have a proper crash log. No need to put ugly swiftlint disable lines, if you can handle it clean with a guard + error message and as a side effect create a nicer crash log.